File: js/24_grafo_direcionado.js
const Graph = require('./lib/graph.js');

let familia = new Graph(true);  // Grafo direcionado

familia.addEdge("Temístocles", "Gercina", "marido de");
familia.addEdge("Gercina", "Temístocles", "esposa de");
console.log(familia.toString());

familia.addEdge("Jerusa", "Temístocles", "filha de");
familia.addEdge("Jerusa", "Gercina", "filha de");
familia.addEdge("Temístocles", "Jerusa", "pai de");
familia.addEdge("Gercina", "Jerusa", "mãe de");
console.log(familia.toString());

File: python/24_grafo_direcionado.py
from lib.graph import Graph

familia = Graph(True)   # Grafo direcionado

familia.add_edge("Temístocles", "Gercina", "marido de")
familia.add_edge("Gercina", "Temístocles", "esposa de")
print(familia)

familia.add_edge("Jerusa", "Temístocles", "filha de")
familia.add_edge("Jerusa", "Gercina", "filha de")
familia.add_edge("Temístocles", "Jerusa", "pai de")
familia.add_edge("Gercina", "Jerusa", "mãe de")
print(familia)
